src/core/ext/xds/xds_client.cc

```diff
@@ -1417,10 +1417,10 @@ void XdsClient::Orphan() {
 }
 
 RefCountedPtr<XdsClient::ChannelState> XdsClient::GetOrCreateChannelStateLocked(
-    const XdsBootstrap::XdsServer& server) {
+    const XdsBootstrap::XdsServer& server, const char* reason) {
```

Comment:
check if absl::string_view is better here

Reply:
In this case, the only place this value will be used is in `gpr_log()`, which takes `const char*`, so using `absl::string_view` would require dynamic allocation (`std::string(reason).c_str()`) every time we log the string.